Assets/PatchKit Patcher/Scripts/AppUpdater/Commands/InstallContentCommand.cs

```diff
@@ -187,30 +189,44 @@ private void InstallFile(SourceFile sourceFile, CancellationToken cancellationTo
                 DebugLogger.LogFormat("Destination file {0} already exists, removing it.", destinationFilePath);
                 FileOperations.Delete(destinationFilePath, cancellationToken);
             }
-
-            FileOperations.Move(sourceFile.FullPath, destinationFilePath, cancellationToken);
+#if UNITY_STANDALONE_WIN
+            if (destinationFilePath.Length > 259)
+            {
+                throw new InstallerException(string.Format("Cannot install file {0}, path is too length.", destinationFilePath)); 
+            }
+#endif
+            FileOperations.Move(sourceFile.FullHashPath, destinationFilePath, cancellationToken);
             _localMetaData.RegisterEntry(sourceFile.Name, _versionId);
-        }
+            }
 
         struct SourceFile
         {
             public string Name { get; private set; }
+            public string Hash { get; private set; }
             private string _suffix;
             private string _root;
 
             public string FullPath { get { return Path.Combine(_root, Name + _suffix); } }
 
-            public SourceFile(string name, string root, string suffix)
+            public string FullHashPath { get { return Path.Combine(_root, Hash + _suffix); } }
+
+            public SourceFile(string name, string root, string suffix, string hash)
             {
```

Comment:
Can we add some assertions here with null checks? I believe Tomasz wrote a class for it, but it's not used too often.

Reply:
done